ansible/roles/test/tasks/bgp_multipath_relax.yml

```diff
@@ -0,0 +1,110 @@
+############################################################################
+### playbook to test leaf router BGP multipath relex
+### There are presumptions for this test that for now:
+#       1) all pre-configured vips are from the same asn and
+#       2) each prefix is "multipath" for leaf router, so we won't validate
+#          if one prefix from vips is multipathed when we select prefix to test
+#       3) we we only test vips advertised ipv4 prefixes
+############################################################################ 
+
+- fail: msg="please provide testbed_type for bgp_multipath_relax test"
+  when: testbed_type is not defined 
+
+- fail: mgs="This test only works for leaf routers as DUT in topology, t1, t1-lag, t1-lag-64"
+  when: testbed_type not in ['t1', 't1-lag', 't1-64-lag']
+
+- name: Gathering minigraph facts about the device
+  minigraph_facts: host={{ inventory_hostname }}
+  connection: local
+
+- name:  Find all V4 bgp neighbors from minigraph
+  set_fact:
+    bgp_v4nei: "{{ bgp_v4nei | default({}) | combine({ item['name']: item['addr'] }) }}"
+  with_items: minigraph_bgp
+  when: "'::' not in item['addr']"
+
+- include_vars: "vars/topo_{{ testbed_type }}.yml"
+
+- name: get all t0 routers name which has vips defined
+  set_fact:
+    dut_t0_nei: "{{ dut_t0_nei|default([]) + [ item ] }}"
+  with_items: "{{ topology['VMs'] }}"
+  when:
+    - "'T0' in item"
+    - "'vips' in configuration[item].keys()"
+
+- name: get all t2 routers names
+  set_fact:
+    dut_t2_nei: "{{ dut_t2_nei|default([]) + [ item ] }}"
+  with_items: "{{ topology['VMs'] }}"
+  when: '"T2" in item'
+
+- name: find all vips prefixes
+  set_fact:
+     vips_prefixes: "{{ vips_prefixes | default([])  + configuration[item]['vips']['ipv4']['prefixes'] | unique}}"
+  with_items: "{{ dut_t0_nei }}"
+
+- name: remove duplicate entries
+  set_fact:
+     vips_prefixes: "{{ vips_prefixes | unique }}"
```

Comment:
why do we need another unique as we did the unique in line 44 already?

Reply:
you are correct, this is duplicated, there were a syntax error for filter which didn't apply unique